Plugins/Public/docking_modules/Utilities.cpp

```diff
@@ -0,0 +1,306 @@
+{
+	string path;
+	path.reserve(dataPath.size() + 45);
+
+	// Get account directory name
+	_GetFLName GetFLName = (_GetFLName)((char*)hModServer + 0x66370);
+	char szDir[12];
+	GetFLName(szDir, acc->wszAccID);
+
+	path = dataPath + "\\Accts\\MultiPlayer\\" + szDir + "\\" + charFile + ".fl";
+
+	map<string, vector<string>> variables;
+	variables["name"] = vector<string>();
+	ReadFLFile(variables, path);
+
+	wstring charname;
+	if (!variables["name"].empty())
+		charname = EncodeWStringFromStringOfBytes(variables["name"][0]);
+
+	return charname;
+}
+
+bool CheckIfExists(string &path)
+{
+	struct stat buffer;
+	return (stat(path.c_str(), &buffer) == 0);
+}
+
+// Get path to original FL account for this charname.
+string GetFLAccPath(wstring &charname)
+{
+	string path;
+	path.reserve(dataPath.size() + 45);
+
+	char datapath[MAX_PATH];
+	GetUserDataPath(datapath);
+
+	CAccount *acc = HkGetAccountByCharname(charname);
+
+	// Get account directory name.
+	_GetFLName GetFLName = (_GetFLName)((char*)hModServer + 0x66370);
+	char szDir[12];
+	GetFLName(szDir, acc->wszAccID);
+
+	// Get character file name.
+	char szFile[12];
+	GetFLName(szFile, charname.c_str());
+
+	path = dataPath + "\\Accts\\MultiPlayer\\" + szDir + "\\" + szFile + ".fl";
+	return path;
+}
+
+// Removes all specifil lines from file, adds specific lines to end of file, or replaces one line with few another if need to insert lines into specific position. Returns result of check if file exists.
+void EditFLFile(vector<string> &linesToDelete, map<string, vector<string>> &linesToReplace, vector<string> &hookExtLinesToAdd, vector<string> &hookExtLinesToDelete, string &path)
+{
+	string tempPath = path + "-temp";
+	ofstream newFile(tempPath);
+	ifstream file(path);
+
+	// Copy all lines to new file with some editions.
+	if (file.is_open())
+	{
+		string line;
+
+	begin:
+		while (getline(file, line))
+		{
+			// Do not scan for hookext data.
+			if (boost::algorithm::starts_with(line, "[flhook]"))
+			{
+				newFile << line << endl;
+				break;
+			}
+
+			for (string &del : linesToDelete)
+				if (boost::algorithm::starts_with(line, del))
+					goto begin;
+			for (auto &pair : linesToReplace)
+			{
+				if (boost::algorithm::starts_with(line, pair.first))
+				{
+					for (string &replacement : pair.second)
+						newFile << replacement << endl;
+
+					goto begin;
+				}
+			}
+
+			newFile << line << endl;
+		}
+
+	hbegin:
+		while (getline(file, line))
+		{
+			for (string &del : hookExtLinesToDelete)
+				if (boost::algorithm::starts_with(line, del))
+					goto hbegin;
+
+			newFile << line << endl;
+		}
+
+		file.close();
+
+		// Add hookext data.
+		for (string &add : hookExtLinesToAdd)
+			newFile << add << endl;
+
+		newFile.close();
+		remove(path.c_str());
+		rename(tempPath.c_str(), path.c_str());
+	}
+}
+
+// My own function to read specific values from FL save file without ini headers. Gets them in raw string format. Returns result of check if file exists.

```

Comment:
let's document that this will populate the map with all lines that *begun* with a key we set in the map

Reply:
also, returns void.

Reply:
Documented, fixed.